### PR TITLE
(fix) tg4x Misc (See desc)

### PR DESCRIPTION
**Pull request by mcrosson**
Various fixes / cleanup

* compile warnings
* keymaps updated to match upstream
* add readme
* add underglow

### CODE REVIEW

**Comment by innovaker** [comment]
This is arguably a good example of the difference between a new feature PR and an improvements or fixes PR.

For a new feature PR, bundling it all together is (usually) fine because it's all part of the same package.  It's atomic in itself.

For further fixes and improvements however, it's better to use atomic commits.  So each of those bullet points would probably be a different commit.  Same PR, sure, but different commits.  It makes reviewing and tracing easier and it also gives you more "room" for detailing context or reasoning within commit messages.  It's simple to do, you just commit as you go - but it's a habit that doesn't seem to be intuitive to most developers.  Doing so actually helps develop general best practices because you learn to avoid making lots of changes at once, which is critical for developing complex systems or tracing bugs.

Perhaps look at it from this perspective (one of many potential scenarios) ... say your PR accidentally breaks some random person's keyboard.  So this poor soul starts digging through the code and finds the offending line in `tg4x.keymap`.  They look at the commit for that line to find out why it was changed.   They read: `(fix) TG4X - Keymaps ; underglow ; build warnings` and are left with these questions ...
- Why was this "fixed"?
- Why are all of these changes seemingly connected/related?  Do I need to worry about them also?
- #help!?

If the commits are atomic, the poor soul has at least got a fighting chance of figuring it out themselves.  The same applies to reviewers and maintainers.

File: app/boards/shields/tg4x/Kconfig.defconfig [hunk 1]
```diff
@@ -4,6 +4,6 @@
 if SHIELD_TG4X
 
 config ZMK_KEYBOARD_NAME
-    default "TG4X"
+    default "tg4x"
```
Comment (Nicell) [inline app/boards/shields/tg4x/Kconfig.defconfig:7]:
```suggestion
    default "TG4x"
```
The silkscreen on the board shows this style of capitalization.

File: app/boards/shields/tg4x/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# TG4X
```
Comment (Nicell) [inline app/boards/shields/tg4x/README.md:1]:
```suggestion
# TG4x
```

File: app/boards/shields/tg4x/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+Standard setup for the [TG4X](https://github.com/MythosMann/tg4x/) 40% keyboard.
```
Comment (Nicell) [inline app/boards/shields/tg4x/README.md:3]:
```suggestion
Standard setup for the [TG4x](https://github.com/MythosMann/tg4x/) 40% keyboard.
```

File: app/boards/shields/tg4x/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+The `keymap` directory contains the keyboard layout and were created by [Keyboard Layout Editor](http://www.keyboard-layout-editor.com/). Please note: the zmk keymap is slightly adjusted from the default shown on the main TG4X pages and QMK.
```
Comment (Nicell) [inline app/boards/shields/tg4x/README.md:5]:
```suggestion
The `keymap` directory contains the keyboard layout and were created by [Keyboard Layout Editor](http://www.keyboard-layout-editor.com/). Please note: the ZMK keymap is slightly adjusted from the default shown on the main TG4x pages and QMK.
```

File: app/boards/shields/tg4x/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+## Board Revision and Layout Notes
+
+This TG4X implementation is for...
```
Comment (Nicell) [inline app/boards/shields/tg4x/README.md:9]:
```suggestion
This TG4x implementation is for...
```

File: app/boards/shields/tg4x/tg4x.overlay [hunk 6]
```diff
@@ -1,56 +1,55 @@
...
+	default_transform: keymap_transform_0 {
+		compatible = "zmk,matrix-transform";
+		rows = <8>;
+        columns = <7>;
 
+        map = <
+RC(0,0) RC(0,1) RC(0,2) RC(0,3) RC(0,4) RC(0,5) RC(0,6) RC(4,0) RC(4,1) RC(4,2) RC(4,3) RC(4,4) RC(4,5)
+RC(1,0) RC(1,1) RC(1,2) RC(1,3) RC(1,4) RC(1,5) RC(1,6) RC(5,0) RC(5,1) RC(5,2) RC(5,3) RC(5,4)
+RC(2,0) RC(2,1) RC(2,2) RC(2,3) RC(2,4) RC(2,5) RC(2,6) RC(6,0) RC(6,1) RC(6,2) RC(6,4)
+RC(3,0) RC(3,1) RC(3,2) RC(3,4) RC(3,5) RC(7,1) RC(7,2) RC(7,3) RC(7,4)
```
Comment (Nicell) [inline app/boards/shields/tg4x/tg4x.overlay:47]: Could this transform possibly have whitespace to show the size of keys better?

**Review by Nicell — CHANGES_REQUESTED** [review]
Some naming thoughts and comments. Otherwise the changes seem pretty reasonable.

**Review by Nicell — APPROVED** [review]

File: app/boards/shields/tg4x/boards/nice_nano.conf [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# Enable underglow
```
Comment (innovaker) [inline app/boards/shields/tg4x/boards/nice_nano.conf:1]:
```suggestion
# Copyright (c) 2020 The ZMK Contributors
# SPDX-License-Identifier: MIT

# Enable underglow
```

Comment (innovaker) [inline app/boards/shields/tg4x/boards/nice_nano.conf:1]:
```suggestion
# Copyright (c) 2020 The ZMK Contributors
# SPDX-License-Identifier: MIT

# Enable underglow
```

**Review by innovaker — CHANGES_REQUESTED** [review]
You have a mix of tabs and spaces in `tg4x.overlay`.

**Comment by mcrosson** [comment]
All changes/updates/suggestions should now be part of this PR

**Review by innovaker — APPROVED** [review]

**Pull request merged**